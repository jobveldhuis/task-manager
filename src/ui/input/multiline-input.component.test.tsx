import { fireEvent, render } from "@testing-library/react-native";
import { MultilineInput } from "@/ui/input/multiline-input.component";
import { TextInput } from "react-native";

describe("MultilineInput", () => {
  const handleChange = jest.fn();

  const component = (
    <MultilineInput
      value="Test"
      onChange={handleChange}
      maxLength={40}
      numberOfLines={3}
    />
  );

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("should match snapshot for a multiline input", () => {
    expect(render(component).toJSON()).toMatchSnapshot();
  });

  it("should fire an onChange with the new value of text", () => {
    const { UNSAFE_getByType: getByType } = render(component);

    const input = getByType(TextInput);
    expect(input).toBeTruthy();

    expect(handleChange).not.toHaveBeenCalled();
    fireEvent(input, "changeText", "The Wizard of Oz");
    expect(handleChange).toHaveBeenCalledWith("The Wizard of Oz");
  });

  it("should not fire the onChange when the amount of maximum lines is exceeded", () => {
    const { UNSAFE_getByType: getByType } = render(component);

    const input = getByType(TextInput);
    expect(input).toBeTruthy();

    expect(handleChange).not.toHaveBeenCalled();
    fireEvent(input, "changeText", "The\nWizard\nof\nOz\nis\ngreat");
    expect(handleChange).not.toHaveBeenCalled();
  });
});
