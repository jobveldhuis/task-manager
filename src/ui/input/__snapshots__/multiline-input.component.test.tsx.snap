// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot for a new password input 1`] = `
<View>
  <TextInput
    autoComplete="new-password"
    autoCorrect={false}
    inputMode="text"
    onChangeText={[Function]}
    secureTextEntry={true}
    style={
      {
        "borderColor": "#000C49",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#000C49",
        "fontSize": 16,
        "paddingHorizontal": 4,
        "paddingVertical": 8,
        "width": "100%",
      }
    }
    value="VerySecret123!"
  />
</View>
`;

exports[`Input should match snapshot for a password input 1`] = `
<View>
  <TextInput
    autoComplete="current-password"
    autoCorrect={false}
    inputMode="text"
    onChangeText={[Function]}
    secureTextEntry={true}
    style={
      {
        "borderColor": "#000C49",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#000C49",
        "fontSize": 16,
        "paddingHorizontal": 4,
        "paddingVertical": 8,
        "width": "100%",
      }
    }
    value="VerySecret123!"
  />
</View>
`;

exports[`Input should match snapshot for a text input 1`] = `
<View>
  <TextInput
    autoComplete="off"
    autoCorrect={false}
    inputMode="text"
    onChangeText={[Function]}
    secureTextEntry={false}
    style={
      {
        "borderColor": "#000C49",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#000C49",
        "fontSize": 16,
        "paddingHorizontal": 4,
        "paddingVertical": 8,
        "width": "100%",
      }
    }
    value="Test"
  />
</View>
`;

exports[`Input should match snapshot for an email input 1`] = `
<View>
  <TextInput
    autoComplete="email"
    autoCorrect={false}
    inputMode="email"
    onChangeText={[Function]}
    secureTextEntry={false}
    style={
      {
        "borderColor": "#000C49",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#000C49",
        "fontSize": 16,
        "paddingHorizontal": 4,
        "paddingVertical": 8,
        "width": "100%",
      }
    }
    value="foo@bar.com"
  />
</View>
`;

exports[`MultilineInput should match snapshot for a multiline input 1`] = `
<View>
  <TextInput
    autoComplete="off"
    autoCorrect={false}
    inputMode="text"
    onChangeText={[Function]}
    secureTextEntry={false}
    style={
      {
        "borderColor": "#000C49",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#000C49",
        "fontSize": 16,
        "paddingHorizontal": 4,
        "paddingVertical": 8,
        "width": "100%",
      }
    }
    value="Test"
  />
</View>
`;
